pkg/sysinfo/cloud/cloud.go

```diff
@@ -58,7 +58,7 @@ type Harvester interface {
 // Detector is used to detect the cloud type on which the instance is running
 // and can be queried in order to get the information needed.
 type Detector struct {
-	sync.Mutex
+	sync.RWMutex
```

Comment:
a var would be better to avoid exposing the mutex api

Reply:
I was worried making a bigger change to `cloud.go`. I'm definitely with you on this though as I think a var would be better than exposing mutex api.